lumigator/python/mzai/backend/backend/tests/conftest.py

```diff
@@ -53,6 +69,58 @@ def format_dataset(data: list[list[str]]) -> str:
     return buffer.read()
 
 
+def _wait_for_job(client, job_id: UUID) -> bool:
+    succeeded = False
+    timed_out = True
+    for _ in range(1, MAX_POLLS):
+        get_job_response = client.get(f"/jobs/{job_id}")
+        assert get_job_response.status_code == 200
+        get_job_response_model = JobResponse.model_validate(get_job_response.json())
+        if get_job_response_model.status == JobStatus.SUCCEEDED.value:
+            succeeded = True
+            timed_out = False
+            break
+        if get_job_response_model.status == JobStatus.FAILED.value:
+            succeeded = False
+            timed_out = False
+            break
+        time.sleep(POLL_WAIT_TIME)
+    if timed_out:
+        raise Exception("Job poll timed out")
+    return succeeded
+
+
+def _wait_for_experiment(client, experiment_id: UUID) -> bool:
+    succeeded = False
+    timed_out = True
+    for _ in range(1, MAX_POLLS):
+        get_experiment_response = client.get(f"/experiments_new/{experiment_id}")
+        assert get_experiment_response.status_code == 200
+        get_experiment_response_model = JobResponse.model_validate(get_experiment_response.json())
+        if get_experiment_response_model.status == JobStatus.SUCCEEDED.value:
+            succeeded = True
+            timed_out = False
+            break
+        if get_experiment_response_model.status == JobStatus.FAILED.value:
+            succeeded = False
+            timed_out = False
+            break
+        time.sleep(POLL_WAIT_TIME)
+    if timed_out:
+        raise Exception("Experiment poll timed out")
+    return succeeded
+
+
+@pytest.fixture
+def wait_for_job(local_client: TestClient) -> Callable[[UUID], bool]:
+    return lambda job_id: _wait_for_job(local_client, job_id)
+
+
+@pytest.fixture
+def wait_for_experiment(local_client: TestClient) -> Callable[[UUID], bool]:
```

Comment:
we shouldn't need a separate fixture here, we should be able to make the current code a fixture

Reply:
We will leave utility functions in the conftest and import them in the tests, as agreed.